river/Server.zig

```diff
@@ -66,6 +66,7 @@ new_layer_surface: wl.Listener(*wlr.LayerSurfaceV1),
 
 xwayland: if (build_options.xwayland) *wlr.Xwayland else void,
 new_xwayland_surface: if (build_options.xwayland) wl.Listener(*wlr.XwaylandSurface) else void,
+runtime_xwayland: bool,
```

Comment:
I think we should make `Server.xwayland` have type `?*wlr.Xwayland` rather than using a separate bool field. I think it would be a lot less error prone than leaving the pointer undefined and needing to remember to check a separate bool every time before accessing it.

Reply:
The only examples I've seen where an optional makes sense are the `Server.xwayland` and `Root.layers.xwayland_override_redirect` struct fields since these are global and we can avoid creating the objects they point to if xwayland is disabled at runtime.

The other examples you've listed don't seem like they would gain anything by using a optional type.